Covid19Radar/Covid19Radar/Services/TestNativeImplementation.cs

```diff
@@ -4,79 +4,198 @@
-
-			Task<IEnumerable<ExposureInfo>> GetInfo()
-			{
-				var info = new List<ExposureInfo>
-				{
-					new ExposureInfo (DateTime.UtcNow.AddDays(-10), TimeSpan.FromMinutes(15), 65, 5, RiskLevel.Medium),
-					new ExposureInfo (DateTime.UtcNow.AddDays(-11), TimeSpan.FromMinutes(5), 40, 3, RiskLevel.Low),
-				};
-				return Task.FromResult<IEnumerable<ExposureInfo>>(info);
-			}
-
-			return Task.FromResult<(ExposureDetectionSummary, Func<Task<IEnumerable<ExposureInfo>>>)>((summary, GetInfo));
-		}
-
-		static TemporaryExposureKey GenerateRandomKey(int daysAgo)
-		{
-			var buffer = new byte[16];
-			random.NextBytes(buffer);
-
-			return new TemporaryExposureKey(
-				buffer,
-				DateTimeOffset.UtcNow.AddDays(-1 * daysAgo),
-				TimeSpan.FromMinutes(random.Next(5, 120)),
-				(RiskLevel)random.Next(1, 8));
-		}
-	}
+    public class MockCommonUtils
+    {
+        public string CdnUrlBase { get => AppSettings.Instance.CdnUrlBase; }
+        public string ApiUrlBase { get => AppSettings.Instance.ApiUrlBase; }
+
+        public bool IsDownloadRequired()
+                => Regex.IsMatch(CdnUrlBase, @"^https://.*\..*\..*/$");
+
+        public bool IsDirectInput()
+        => Regex.IsMatch(CdnUrlBase, @"^(\d+,)+\d+,*$");
+
+
+        private ushort NumberEndofSentence(string url)
+        {
+            Match match = Regex.Match(url, @"(?<d>\d+)$");
+            ushort number = 0;
+            if (match.Success)
+            {
+                number = Convert.ToUInt16(match.Groups["d"].Value);
+            }
+            return (number);
+        }
+        public List<string> GetCreatedTimes()
+            => CdnUrlBase.Split(",").ToList();
+        public ushort GetTekListDataType()
+        => NumberEndofSentence(CdnUrlBase);
+        public string[] GetApiUrlSegment()
+        {
+            // "url/api" -> { "url/api", "", "" }
+            // "url/base/api/register1/diagnosis2" -> { "url/base/api", "/register1", "/diagnosis2" } 
+            // "url/api1/r1/d2" -> { "url/api1", "/r1", "/d2" } 
+            // "url/api1/d2/r1" -> { "url/api1", "/r1", "/d2" } 
+            var url = ApiUrlBase;
+            var r = new Regex("/r(egister)?[0-9]+");
+            var d = new Regex("/d(iagnosis)?[0-9]+");
+            var urlRegister = r.Match(url).Value;
+            url = r.Replace(url, "");
+            var urlDiagnosis = d.Match(url).Value;
+            url = d.Replace(url, "");
+            var urlApi = url;
+            return (new string[] { urlApi, urlRegister, urlDiagnosis });
+        }
+        public ushort GetDiagnosisDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[2]);
+        public ushort GetRegisterDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[1]);
+        public ushort GetApiDataType()
+        => NumberEndofSentence(GetApiUrlSegment()[0]);
+        public bool IsDirectInputApi()
+        => Regex.IsMatch(GetApiUrlSegment()[0], @"^(\d+,)+\d+,?$");
```

Comment:
style nitpick indent

```suggestion
        public ushort GetDiagnosisDataType()
            => NumberEndofSentence(GetApiUrlSegment()[2]);
        public ushort GetRegisterDataType()
            => NumberEndofSentence(GetApiUrlSegment()[1]);
        public ushort GetApiDataType()
            => NumberEndofSentence(GetApiUrlSegment()[0]);
        public bool IsDirectInputApi()
            => Regex.IsMatch(GetApiUrlSegment()[0], @"^(\d+,)+\d+,?$");
```